### PR TITLE
Refactor error handling logic; Add generic client for CRD; Add ctx as parameter for all inner handlers

### PR DESCRIPTION
**Pull request by PowerfooI**
## Summary

1. Create clients for every CRD type with the help of generics. They simplify defining new business logic greatly.
2. Wrap inner handlers with a  function ErrorWrapper (abbr. W), which handles all errors in it. So that we do not need SendResponse manually at all.
3. Add ctx as parameter for all inner handlers
4. Fix some tiny bugs like sending response more than once
5. Implement backup-related APIs

### CODE REVIEW

File: distribution/dashboard/pkg/errors/error.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package errors
+
+import (
+	"fmt"
+	"net/http"
+)
+
+type oberror struct {
```
Comment (chris-sun-star) [inline distribution/dashboard/pkg/errors/error.go:8]: The error is actually HTTP status, call it oberror is misleading

File: distribution/dashboard/internal/handler/wrapper.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/oceanbase/oceanbase-dashboard/internal/model/response"
+	"github.com/oceanbase/oceanbase-dashboard/pkg/errors"
+)
+
+type Handler[T any] func(c *gin.Context) (T, error)
+
+func W[T any](h Handler[T]) gin.HandlerFunc {
```
Comment (chris-sun-star) [inline distribution/dashboard/internal/handler/wrapper.go:13]: Maybe Wrap is better, W is too short

File: distribution/dashboard/internal/handler/metric_handler.go [hunk 3]
```diff
@@ -22,23 +24,21 @@ import (
...
-			logHandlerError(c, err)
-			SendInternalServerErrorResponse(c, nil, err)
-		} else {
-			SendSuccessfulResponse(c, metricClasses)
-		}
+		return nil, oberr.NewBadRequest(err.Error())
 	}
+	metricClasses, err := metric.ListMetricClasses(scope, language)
+	if err != nil {
+		logHandlerError(c, err)
```
Comment (chris-sun-star) [inline distribution/dashboard/internal/handler/metric_handler.go:38]: logHandlerError can be moved into wrapper, thus we can log all the errors in one place, all the handlers only need to return the error

File: distribution/dashboard/pkg/errors/error.go (hunk 1 above)
Comment (PowerfooI) [reply distribution/dashboard/pkg/errors/error.go]: OK, it's true. I have renamed it to `buzerror`.

File: distribution/dashboard/internal/handler/metric_handler.go (hunk 3 above)
Comment (PowerfooI) [reply distribution/dashboard/internal/handler/metric_handler.go]: Good idea! I have removed all loggings in specific handlers and moved error logging logic into the handler wrapper.

**Review by chris-sun-star — APPROVED** [review]

**Pull request merged**